content/en/docs/concepts/signals/baggage.md

```diff
@@ -52,23 +55,35 @@ then this risk may not apply, but keep in mind that downstream services could
 propagate Baggage outside your network.
 
 Also, there are no built-in integrity checks to ensure that Baggage items are
-yours, so exercise caution when retrieving them.
+yours, so exercise caution when reading them.
 
-## Baggage is not the same as Span attributes
+## Baggage is not the same as attributes
 
-One important thing to note about Baggage is that it is not a subset of the
-[Span Attributes](/docs/concepts/signals/traces/#attributes). When you add
-something as Baggage, it does not automatically end up on the Attributes of the
-child system’s spans. You must explicitly take something out of Baggage and
-append it as Attributes.
+An important thing to note about Baggage is that it is a separate key:value
+store and is unassociated with attributes on spans, metrics, or logs.
 
-For example, in .NET you might do this:
+To add Baggage entries to attributes, you need to explicitly read that data and
+add it. For example, in .NET you might do this:
 
 ```csharp
 var accountId = Baggage.GetBaggage("AccountId");
 Activity.Current?.SetTag("AccountId", accountId);
 ```
 
+Because one of the most common use cases for Baggage is to add data to
+[Span Attributes](/docs/concepts/signals/traces/#attributes) across a whole
+trace, several languages have Baggage Span Processors that automatically read
+Baggage values and add them as attributes for you.
+
+## Baggage is only for observability data
+
+Although Baggage does support propagating arbitrary key:value pairs across
+services, it is not a replacement for programmatic data flow. Because Baggage is
+stored in HTTP headers, not only does this mean the data is available for anyone
+who can read those headers to consume, but it would be a poor choice to
+propagate general program data given the practical limits HTTP servers place on
+header lengths.
```

Comment:
I like this section. To make the disadvantage maybe even more apparent you could give an example where headers get dropped by a proxy inbetween services, which then **only** affects observability but not the functionality of your app

Reply:
Done